File: src/main/java/io/github/enummapper/compiler/CompilerReferences.java
package io.github.enummapper.compiler;

import java.util.Collection;
import org.jetbrains.kotlin.backend.common.extensions.IrPluginContext;
import org.jetbrains.kotlin.ir.symbols.IrClassSymbol;
import org.jetbrains.kotlin.ir.symbols.IrSimpleFunctionSymbol;
import org.jetbrains.kotlin.name.CallableId;
import org.jetbrains.kotlin.name.ClassId;

final class CompilerReferences {
    private CompilerReferences() {
    }

    static Collection<IrSimpleFunctionSymbol> functions(
            IrPluginContext context, CallableId callableId) {
        return context.referenceFunctions(callableId);
    }

    static IrClassSymbol classSymbol(IrPluginContext context, ClassId classId) {
        return context.referenceClass(classId);
    }
}

File: src/main/kotlin/cz/pavelflegr/remap/compiler/CompilerReferences.kt
@file:Suppress("DEPRECATION", "DEPRECATION_ERROR")

package cz.pavelflegr.remap.compiler

import org.jetbrains.kotlin.backend.common.extensions.IrPluginContext
import org.jetbrains.kotlin.ir.symbols.IrClassSymbol
import org.jetbrains.kotlin.ir.symbols.IrSimpleFunctionSymbol
import org.jetbrains.kotlin.name.CallableId
import org.jetbrains.kotlin.name.ClassId

internal object CompilerReferences {
    fun functions(
        context: IrPluginContext,
        callableId: CallableId,
    ): Collection<IrSimpleFunctionSymbol> = context.referenceFunctions(callableId)

    fun classSymbol(context: IrPluginContext, classId: ClassId): IrClassSymbol =
        context.referenceClass(classId)!!
}
